functions/New-DbaSqlParameter.ps1

```diff
@@ -0,0 +1,215 @@
+
+    .PARAMETER Precision
+        Sets the maximum number of digits used to represent the Value property.
+
+    .PARAMETER Scale
+        Sets the number of decimal places to which Value is resolved.
+
+    .PARAMETER Size
+        Sets the maximum size, in bytes, of the data within the column.
+
+    .PARAMETER SourceColumn
+        Sets the name of the source column mapped to the DataSet and used for loading or returning the Value.
+
+    .PARAMETER SourceColumnNullMapping
+        Sets a value which indicates whether the source column is nullable. This allows SqlCommandBuilder to correctly generate Update statements for nullable columns.
+
+    .PARAMETER SourceVersion
+        Sets the DataRowVersion to use when you load Value.
+
+    .PARAMETER SqlDbType
+        Sets the SqlDbType of the parameter.
+
+    .PARAMETER SqlValue
+        Sets the value of the parameter as an SQL type.
+
+    .PARAMETER TypeName
+        Sets the type name for a table-valued parameter.
+
+    .PARAMETER UdtTypeName
+        Sets a string that represents a user-defined type as a parameter.
+
+    .PARAMETER Value
+        Sets the value of the parameter.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Query
+        Author: Chrissy LeMaire (@cl), netnerds.net
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaSqlParameter
+
+    .EXAMPLE
+        PS C:\> New-DbaSqlParameter -ParameterName json_result -SqlDbType NVarChar -Size -1 -Direction Output
+
+        Creates a SqlParameter object that can be used with Invoke-DbaQuery
+
+    .EXAMPLE
+        PS C:\> $output = New-DbaSqlParameter -ParameterName json_result -SqlDbType NVarChar -Size -1 -Direction Output
+        PS C:\> Invoke-DbaQuery -SqlInstance localhost -Database master -CommandType StoredProcedure -Query my_proc -SqlParameters $output
+        PS C:\> $output.Value
+
+        Creates an output parameter and uses it to invoke a stored procedure.
+    #>
+    [CmdletBinding()]
+    param(
+        [ValidateSet("None", "IgnoreCase", "IgnoreNonSpace", "IgnoreKanaType", "IgnoreWidth", "BinarySort2", "BinarySort")]
+        [string]$CompareInfo,
+        [ValidateSet("AnsiString", "Binary", "Byte", "Boolean", "Currency", "Date", "DateTime", "Decimal", "Double", "Guid", "Int16", "Int32", "Int64", "Object,
+        SByte", "Single", "String", "Time", "UInt16", "UInt32", "UInt64", "VarNumeric", "AnsiStringFixedLength", "StringFixedLength", "Xml,
```

Comment:
not sure this is indented correctly

Reply:
you were right! thanks, that was super messed up.